src/app/context/AccountContext.tsx

```diff
@@ -75,15 +111,15 @@ export function AccountProvider({ children }: { children: React.ReactNode }) {
     });
   };
 
-  const setAccountId = (id: string) => setAccount({ id });
+  const setAccountId = async (id: string) => {
+    setAccount({ id });
+  };
 
-  const updateFiatValue = useCallback(
-    async (balance: string | number) => {
-      const fiat = await getFormattedFiat(balance);
-      setFiatBalance(fiat);
-    },
-    [getFormattedFiat]
-  );
+  const refetchAccountInfo = async () => {
+    if (typeof mutate !== "function") return;
+    const data = await mutate();
+    setAccountData(account?.id, data);
```

Comment:
I still don't think `setAccountData(account?.id, data);`  is needed. If you call mutate it will refetch the data using SWR and update the SWR cache.

Reply:
Indeed. Now the AccountMenu is updated even though I don't setAccountData. Not sure why it didn't work when I tried initially. Good you persisted.